src/test/e2e/02_component_actions_test.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package test provides e2e tests for Zarf.
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestComponentActions(t *testing.T) {
+	t.Log("E2E: Testing component actions")
+	e2e.setup(t)
+	defer e2e.teardown(t)
+
+	// Note these files will be created in the package directory, not CWD.
+	createArtifacts := []string{
+		"examples/component-actions/test-create-before.txt",
+		"examples/component-actions/test-create-after.txt",
+	}
+	deployArtifacts := []string{
+		"test-deploy-before.txt",
+		"test-deploy-after.txt",
+	}
+	deployWithEnvVarArtifact := "filename-from-env.txt"
+
+	allArtifacts := append(deployArtifacts, createArtifacts...)
+	allArtifacts = append(allArtifacts, deployWithEnvVarArtifact)
+	e2e.cleanFiles(allArtifacts...)
+	defer e2e.cleanFiles(allArtifacts...)
+
+	/* Create */
+	// Try creating the package to test the onCreate actions.
+	stdOut, stdErr, err := e2e.execZarfCommand("package", "create", "examples/component-actions", "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+
+	// Test for package create prepare artifact.
+	for _, artifact := range createArtifacts {
+		require.FileExists(t, artifact)
+	}
+
+	// Test to ensure the deploy scripts are not executed.
+	for _, artifact := range deployArtifacts {
+		require.NoFileExists(t, artifact)
+	}
+
+	/* Deploy */
+	path := fmt.Sprintf("build/zarf-package-component-actions-%s.tar.zst", e2e.arch)
+	// Deploy the simple script that should pass.
+	stdOut, stdErr, err = e2e.execZarfCommand("package", "deploy", path, "--confirm", "--components=on-deploy")
+
+	// Check that the deploy artifacts were created.
+	for _, artifact := range deployArtifacts {
+		require.FileExists(t, artifact)
+	}
+
+	// Deploy the simple action that should fail the timeout.
+	stdOut, stdErr, err = e2e.execZarfCommand("package", "deploy", path, "--confirm", "--components=on-deploy-with-timeout")
+	require.Error(t, err, stdOut, stdErr)
+
+	// Test using a Zarf Variable within the action
+	stdOut, stdErr, err = e2e.execZarfCommand("package", "deploy", path, "--confirm", "--components=on-deploy-with-variable", "-l=trace")
+	require.NoError(t, err, stdOut, stdErr)
+	require.Contains(t, stdOut, "the dog says ruff")
+
+	// Test using dynamic and multiple-variables
+	stdOut, stdErr, err = e2e.execZarfCommand("package", "deploy", path, "--confirm", "--components=on-deploy-with-dynamic-variable,on-deploy-with-multiple-variables", "-l=trace")
+	require.NoError(t, err, stdOut, stdErr)
+	require.Contains(t, stdOut, "the cat says meow")
+	require.Contains(t, stdOut, "the dog says ruff")
+	require.Contains(t, stdOut, "the snake says hiss")
+
+	// Test using environment variables
+	stdOut, stdErr, err = e2e.execZarfCommand("package", "deploy", path, "--confirm", "--components=on-deploy-with-env-var")
+	require.NoError(t, err, stdOut, stdErr)
+	require.FileExists(t, deployWithEnvVarArtifact)
```

Comment:
We should add a test for the onRemove case as well.

Reply:
we won't be able to easily test remove since that will require a cluster that won't exist yet.

Reply:
We could add it to another test that could benefit from some remove behavior potentially. I do worry about not testing it because it is effectively a third of this PR (create, deploy, remove)

Reply:
I mean, it's a 3rd call of the majority same code, that's very different than "one-third of the PR". I'd strongly prefer to keep this test pre-cluster, maybe we could have a second actions test with-cluster or something. Not sure.

Reply:
Yeah that is what I was suggesting, and not 1/3 of code but 1/3 of functionality introduced (which could be taken out easily).

Reply:
I may be a little skittish but it is exactly that that completely disabled SBOMing and external registries in the past when simple if statements at higher levels got flipped.

Reply:
It wouldn't necessarily have to be its own test, but could be something we test as a side effect of another test (potentially just with a comment recording that).

Reply:
And case in point this functionality is partially broken (--set is not a thing on remove and it cannot / does not load variable defaults) 
![image](https://user-images.githubusercontent.com/3977569/214132043-356e56f6-d5a6-4c52-9b1e-6ee712b581ea.png)

Reply:
(here is that test case: https://github.com/defenseunicorns/zarf/tree/jeff-doing-stupid-jeff-things-remove-test)

Reply:
(differs slightly from the run above in that I added --set calls)

Reply:
(we could also store the vars used to deploy or explicitly not support variables here in the docs)

Reply:
in the case of create/remove the variables are only generated by actions, which I think makes sense.

Reply:
I see no problem with this behavior, though I suppose we could add more docs for that, too...even though defined variables are very clearly for package deploy only already